moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityForm.js

```diff
@@ -0,0 +1,283 @@
+  } = useForm({
+    defaultValues: {
+      ...defaultValues,
+    },
+    resolver: yupResolver(activityValidationSchema),
+  });
+
+  /** Formats array of task order objects into value prop of multiselect component */
+  const taskOrderValueHandler = useCallback(
+    (valueArray) => {
+      if (!valueArray) {
+        // handles when saved value is null
+        return [];
+      }
+      const selectedTks = valueArray?.map((tkOption) => tkOption.task_order);
+      return (
+        taskOrderOptions?.filter((option) =>
+          selectedTks.includes(option.value.task_order)
+        ) || []
+      );
+    },
+    [taskOrderOptions]
+  );
+
+  const [mutate, mutationState] = useMutation(
+    activity.id ? UPDATE_WORK_ACTIVITY : ADD_WORK_ACTIVITIY
+  );
+
+  if (errorStatuses || errorTaskOrders || mutationState.error) {
+    console.error(errorStatuses || errorTaskOrders || mutationState.error);
+    return (
+      <Grid container spacing={2}>
+        <Grid item xs={12}>
+          <Alert severity="error">
+            Something went wrong. Refresh the page to try again.
+          </Alert>
+        </Grid>
+      </Grid>
+    );
+  } else if (loadingStatuses || loadingTaskOrders) {
+    return (
+      <Grid container spacing={2}>
+        <Grid item xs={12}>
+          <CircularProgress color="primary" size={20} />
+        </Grid>
+      </Grid>
+    );
+  }
+
+  return (
+    <form
+      onSubmit={handleSubmit((data) =>
+        onSubmitActivity({ data, mutate, onSubmitCallback })
+      )}
+      autoComplete="off"
+    >
+      <Grid container spacing={2}>
+        {!isNewActivity && (
+          <Grid item xs={12}>
+            <FormControl fullWidth>
+              <TextField
+                id="id"
+                disabled
+                fullWidth
+                label="ID"
+                size="small"
+                value={activity.id}
+              />
+            </FormControl>
+          </Grid>
+        )}
+        <Grid item xs={12}>
+          <FormControl fullWidth error={!!formErrors.status_id}>
+            <InputLabel id="status-label" required={true}>
+              Status
+            </InputLabel>
+            <ControlledSelect
+              control={control}
+              id="status"
+              labelId="status-label"
+              name="status_id"
+              label="Status"
+              options={statusOptions}
+              error={formErrors?.status_id}
+              autoFocus={isNewActivity}
+              size="small"
+              required={true}
+            />
+            <FormHelperText>Required</FormHelperText>
+          </FormControl>
+        </Grid>
+        <Grid item xs={12}>
+          <FormControl fullWidth>
+            <ControlledAutocomplete
+              control={control}
+              name="contractor"
+              label="Workgroup/Contractor"
+              options={IMPLEMENTATION_WORKGROUP_OPTIONS}
+              freeSolo
```

Comment:
Seeing how this field is stored in the DB as a makes the free solo option make a lot more sense to me now. This works great 👍 

Looking at all this form validation - should we have max lengths on some of these text fields since they are text type in the DB? Just a passing thought because I don't think anyone in the user group would input a massive string or anything. Just not sure if we've ever talked about it.

Reply:
yes—great call. this has crossed my mind in other places and i have forgotten to follow up.

Reply:
...we should probably add some constraints at the DB level as well, but i'll leave that for another day